### PR TITLE
Certificate: add MarshalDER and ComputeFingerprint

### PR DESCRIPTION
**Pull request by huwcbjones**

### CODE REVIEW

File: cert.go [hunk 1]
```diff
@@ -386,6 +386,19 @@ func (c *Certificate) MarshalPEM() (pem_block []byte, err error) {
 	return io.ReadAll(asAnyBio(bio))
 }
 
+// MarshalDER converts the X509 certificate to DER-encoded format
+func (c *Certificate) MarshalDER() (der_block []byte, err error) {
```
Comment (w-miller) [inline cert.go:390]: nit: don't need named return values here

File: key_test.go [hunk 2]
```diff
@@ -56,6 +56,16 @@ func TestMarshal(t *testing.T) {
 		t.Fatal("invalid cert pem bytes")
 	}
 
+	der, err := cert.MarshalDER()
+	if err != nil {
```
Comment (w-miller) [inline key_test.go:60]: nit: `require.NoError(t, err)` is a bit more concise

File: key_test.go [hunk 3]
```diff
@@ -75,14 +85,14 @@ func TestMarshal(t *testing.T) {
...
 	_ = tls_key
 
-	der, err := key.MarshalPKCS1PrivateKeyDER()
+	pkcs1Der, err := key.MarshalPKCS1PrivateKeyDER()
 	if err != nil {
 		t.Fatal(err)
 	}
 	tls_der := x509.MarshalPKCS1PrivateKey(tls_key)
-	if !bytes.Equal(der, tls_der) {
+	if !bytes.Equal(pkcs1Der, tls_der) {
```
Comment (w-miller) [inline key_test.go:93]: I think you can use `require.Equal` here -  it prints the two values upon failure as well

**Review by w-miller — APPROVED** [review]

File: key_test.go (hunk 2 above)
Comment (huwcbjones) [reply key_test.go:60]: Agreed, but require is not currently used and this matches the current codebase style

File: key_test.go (hunk 3 above)
Comment (huwcbjones) [reply key_test.go:93]: Agreed, but require is not currently used and this matches the current codebase style

File: cert.go (hunk 1 above)
Comment (huwcbjones) [reply cert.go:390]: Agreed, but this matches the current codebase style

**Pull request merged**